fix: kubectl argument too long when updating release cr

arewm opened the pull request:
## Describe your changes

Most of the development and CI has been done assuming smaller Snapshots. When releasing larger snapshots (i.e. 180 components), we start to encounter issues ranging from higher likelihood to hit timeouts, increased memory consumption, and long strings being processed.

This PR is focused on fixing an issue where `update-cr-status` could not process large snapshots, namely because we were passing too much information to `kubectl`.

Partially addresses: [KONFLUX-8821](https://issues.redhat.com//browse/KONFLUX-8821) in conjunction with #1218, #1219, #1220

Assisted-by: Cursor
Signed-off-by: arewm <arewm@users.noreply.github.com>

## Relevant Jira

## Checklist before requesting a review
- [x] I have marked as draft or added `do not merge` label if there's a dependency PR
  - If you want reviews on your draft PR, you can add reviewers or add the `release-service-maintainers` handle if you are unsure who to tag
- [x] My commit message includes `Signed-off-by: My name <email>`
- [x] I read CONTRIBUTING.MD and [commit formatting](CONTRIBUTING.md#commit-message-formatting-and-standards)
- [x] I have run the README.md generator script in `.github/scripts/readme_generator.sh` and verified the results using `.github/scripts/check_readme.sh`

FilipNikolovski reviewed (dismissed):
lgtm apart from the lint errors.

arewm committed:
fix: kubectl argument too long

When patching the Release with a lot of information, we are putting too
much into the command line to be able to be processed. We can use a
patch file instead of passing it all in the command line.

Assisted-by: Cursor
Signed-off-by: arewm <arewm@users.noreply.github.com>

rh-pre-commit.version: 2.3.2
rh-pre-commit.check-secrets: ENABLED

johnbieren commented:
```
⚠️ Warning: Could not get component PR from annotations: {"pac":{"state":"error","error-id":74,"error-message":"74: Access token is unrecognizable by GitHub"},"message":"done"}
Requesting a new configure-pac...
component.appstudio.redhat.com/rhtap-service-push-b8455b79 annotate
Waiting 10 seconds before retry...
Initialization check attempt 58/60...
```
I haven't seen that before

johnbieren commented:
/retest
